image/helpers_test.go

```diff
@@ -130,3 +132,93 @@ func (s *imageSuite) TestDownloadSnap(c *check.C) {
 
 	c.Check(logbuf.String(), check.Matches, `.* DEBUG: Going to download snap "core" `+opts.String()+".\n")
 }
+
+var validGadgetYaml = `
+volumes:
+  vol1:
+    bootloader: grub
+    structure:
+      - name: non-fs
+        type: bare
+        size: 512
+        offset: 0
+        content:
+        - image: non-fs.img
+      - name: ubuntu-seed
+        role: system-seed
+        type: 83,0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        size: 100M
+        filesystem: ext4
+        content:
+         - source: system-seed.efi
+           target: EFI/boot/system-seed.efi
+      - name: structure-name
+        role: system-boot
+        type: 83,0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        size: 100M
+        filesystem: ext4
+        content:
+         - source: grubx64.efi
+           target: EFI/boot/grubx64.efi
+      - name: ubuntu-data
+        role: system-data
+        type: 83,0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        size: 100M
+  vol2:
+    structure:
+      - name: struct2
+        type: 83,0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        size: 100M
+        filesystem: ext4
+        content:
+         - source: foo
+           target: subdir/foo
+`
+
+func (s *imageSuite) TestWriteResolvedContent(c *check.C) {
+	prepareImageDir := c.MkDir()
+	// on uc20 there is a "system-seed" under the <PrepareImageDir>
+	uc20systemSeed := filepath.Join(prepareImageDir, "system-seed")
+	err := os.MkdirAll(uc20systemSeed, 0755)
+	c.Assert(err, check.IsNil)
+
+	// the resolved content is written here
+	gadgetRoot := c.MkDir()
+	snaptest.PopulateDir(gadgetRoot, [][]string{
+		{"meta/snap.yaml", packageGadget},
+		{"meta/gadget.yaml", validGadgetYaml},
+		{"system-seed.efi", "content of system-seed.efi"},
+		{"grubx64.efi", "content of grubx64.efi"},
+		{"foo", "content of foo"},
+		{"non-fs.img", "content of non-fs.img"},
+	})
+	kernelRoot := c.MkDir()
+	model := s.makeUC20Model(nil)
+
+	err = image.WriteResolvedContent(prepareImageDir, gadgetRoot, kernelRoot, model)
```

Comment:
we need a unit test for prepareImageDir being a relative dir too

Reply:
I added that now